### ISSUE TITLE
Failure to download html + problem xblocks on openlearninglibrary.mit.edu

### ISSUE THREAD

**Issue by benoit74** [openzim/openedx#175]
While testing the creation of a ZIM with https://openlearninglibrary.mit.edu/courses/course-v1:OCW+6.042J+2T2019/course/, I encountered many download errors.

It looks like no html and no problem xblocks are downloaded (might need to implement #160 for more details :)).

From what I see, in each case the scrapper tries to retrieve the `student_view_url` but openlearninglibrary.mit.edu server always return a 500 on this URL.

**Comment by benoit74** [comment]
Same issue with another platform (FUN) and another course : https://lms.fun-mooc.fr/courses/course-v1:MinesTelecom+04012+session08/courseware/ba47011e0c9841da83b7fe4836cc6f40/

Looks like the scraper is broken.

**Comment by benoit74** [comment]
For FUN, it looks like removing the `update_csrf_token_in_headers` [here](https://github.com/openzim/openedx/blob/c2e01c23541eb35472d07005f4a915e1bc1f6a05/openedx2zim/instance_connection.py#L95) solves the issue. But it does not work for MIT. So definitely linked to the way we are trying to mimic the UI behavior via headers / cookies to retrieve contents which has changed from one OpenEDX release to the next one.

**Comment by rgaudin** [comment]
I don't get whether this is a general problem (and we can't run any openedx recipe) or if this is a discrete problem that happens with both FUN and MIT (and maybe others)

**Comment by benoit74** [comment]
From my perspective, it mostly looks like a general problem. Most probably linked to some change in openedx, so maybe some old instances might still work.

I can't tell for sure how wide the impact is for us, since there is no more openedx recipes active on the zimfarm, now that we have deactived PHZH recipes.

Do you have some openedx course you would like to test?

**Comment by jo780-full** [comment]
Hello, should the user be logged and enrolled only then it would downloaded???

**Comment by benoit74** [comment]
AFAIK (I'm not an expert of this scraper), user should be enrolled in the course you want to download, yes.

**Comment by lNamelessl** [comment]
Hello, can I work on this issue?@benoit74

**Comment by benoit74** [comment]
@lNamelessl of course, even if this is probably a tough issue. You first have to investigate what the problem is and explain it here. And explain what your plan to fix the issue is. I won't assign the issue until you came up with a plan.

**Comment by clencyc** [comment]
@benoit74 I figured it out. 

<img width="499" height="512" alt="Image" src="https://github.com/user-attachments/assets/7b0c7a7d-0a85-4a2f-8ff3-ca9d7991e777" />

The scraper mimics browser UI behavior by sending specific cookies/CSRF tokens when fetching student_view_url for HTML and problem xblocks. As OpenEdX evolves across releases, the authentication handshake and CSRF token update mechanisms change meaning what worked on one platform/release fails with a 500 on another.
Solution strategy: Rather than tightly coupling the scraper to one specific OpenEdX auth pattern, we should build a resilient, multi-strategy fetcher that:

1. Tries the standard student_view_url fetch first
2. Falls back to alternative content retrieval strategies
3. Handles CSRF token updates per-platform, not globally
4. Gracefully degrades with partial content rather than silent failure

**Comment by benoit74** [comment]
I don't get if the problem is an "auth or alike" problem (like we do not get the CSRF token correctly, ...) or an API problem (like we use API which do not exists or we use them in a wrong manner).
